vignettes/spatialsample.Rmd

```diff
@@ -27,23 +29,110 @@ The resampled objects created by spatialsample can be used in many of the same w
 
 ```{r}
 library(spatialsample)
 
 set.seed(123)
-folds <- spatial_clustering_cv(ames, coords = c("Latitude", "Longitude"), v = 15)
-folds
+cluster_folds <- spatial_clustering_cv(ames_sf, v = 15)
+
+autoplot(cluster_folds)
+```
+
+Our `cluster_folds` object is an `rset` object that contains many resamples or `rsplit` objects in the `splits` column. The resulting partitions do not necessarily contain an equal number of observations:
+
+```{r}
+cluster_folds
+```
+
+But while spatial clustering is _a_ method for spatial cross-validation using spatialsample, it is not the only method available. The other methods are broadly similar, breaking the data into a number of (not necessarily even) folds based on spatial arrangement. 
+
+For instance, the `spatial_block_cv()` function will perform [spatial blocking](https://doi.org/10.1111/ecog.02881) with your data:
+
+```{r}
+set.seed(123)
+block_folds <- spatial_block_cv(ames_sf, v = 15)
+
+autoplot(block_folds)
+```
+
+If you already have a sense of what locations in your data are likely to be closely related, you can also use the `spatial_leave_location_out_cv()` function to perform [leave-location-out cross-validation](https://doi.org/10.1016/j.envsoft.2017.12.001). For instance, we can split the Ames data into folds based on neighborhoods using this function:
+
+```{r}
+set.seed(123)
+location_folds <- 
+  spatial_leave_location_out_cv(
+    ames_sf,
+    group = Neighborhood,
+    v = 15
+  )
+
+autoplot(location_folds)
+```
+
+And last but not least, 
+[spatially-buffered v-fold cross-validation](https://doi.org/10.1111/geb.12161):
+
+```{r}
+set.seed(123)
+buffer_folds <- 
+  spatial_buffer_vfold_cv(
+    ames_sf,
+    v = 15,
+    radius = NULL,
+    buffer = 200
+  )
+
+autoplot(buffer_folds)
 ```
 
-The `folds` object is an `rset` object that contains many resamples or `rsplit` objects in the `splits` column. The resulting partitions do not necessarily contain an equal number of observations.
+Just looking at the default `autoplot()` output, it isn't immediately obvious
+that our `buffer` did anything. However, if we plot each `rsplit` object separately, we can see that we've created a small exclusion buffer around each test (otherwise known as assessment) set:
+
+```{r buffer-gif, animation.hook="gifski", fig.width=7, fig.height=5}
+purrr::walk(buffer_folds$splits, function(x) print(autoplot(x)))
+```
+
+Points inside of this buffer are included in neither the analysis nor assessment sets, making sure you're enforcing a minimum amount of spatial extrapolation when predicting on new data. For more information on this process, check out the [buffering vignette](https://spatialsample.tidymodels.org/dev/articles/buffering.html)!
```

Comment:
For an article (not a vignette) we should put a URL so folks can find it from anywhere, including CRAN. Unfortunately we have to use the `/dev` site because the release one won't exist until after this makes it to CRAN. I'm OK with this if you are @mikemahoney218.

Reply:
Makes sense to me :smile: